File: 백준/Silver/11047. 동전 0/동전 0.py
import sys

N,K=map(int,sys.stdin.readline().split())
coins=[]
count=0

for i in range(N):
    coins.append(int(sys.stdin.readline()))

for i in range(len(coins)-1,-1,-1):    
    if K//coins[i] == 0 :
        continue
    else:
        count+= K//coins[i]
        K -= coins[i] * (K//coins[i]) # rest charge
    if K==0:
        break
        
print(count)

File: 백준/Silver/11047. 동전 0/동전 0.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N;
    long long K;
    
    cin >> N >> K;
    
    vector<int> coins(N);

    for (int i = 0; i < N; i++) {
        cin >> coins[i];
    }

    int count =0;

    for(int i=N-1;i>=0;--i){
        if(coins[i] <= K) {
            count+=K/coins[i];
            K%=coins[i];
        }
    }

    cout << count << '\n';
    return 0;
}
